Testing/integration/tests/Infrastructure/MessagingTests.cs

```diff
@@ -49,25 +49,38 @@ public MessagingTests(ITestOutputHelper output, WebAppFixture fixture) : base(ou
     public override async Task InitializeAsync()
     {
         await base.InitializeAsync();
-        this.receiver = this.Host.Services.GetRequiredService<IMessageReceiver>();
+        this.receiver = this.TestServices.GetRequiredService<IMessageReceiver>();
 
         GlobalConfiguration.Configuration.UsePostgreSqlStorage(
             c => c.UseNpgsqlConnection(
-                this.Host.Services.GetRequiredService<IConfiguration>().GetConnectionString("GatewayConnection")));
+                this.TestServices.GetRequiredService<IConfiguration>().GetConnectionString("GatewayConnection")));
+    }
+
+    public override async Task DisposeAsync()
+    {
+        await base.DisposeAsync();
+        this.cts.Dispose();
     }
 
     private async Task SendMessages(IEnumerable<IEnumerable<TestMessage>> messageGroups, CancellationToken ct)
     {
-        await Parallel.ForEachAsync(
-            messageGroups,
-            ct,
-            async (messages, ct1) =>
-            {
-                this.Output.WriteLine("sending messages");
-                using IServiceScope scope = this.Host.Services.CreateScope();
-                IMessageSender sender = scope.ServiceProvider.GetRequiredService<IMessageSender>();
-                await sender.SendAsync(messages.Select(m => new MessageEnvelope(m, m.SessionId)), ct1);
-            });
+        IMessageSender sender = this.TestServices.GetRequiredService<IMessageSender>();
+
+        foreach (var messages in messageGroups)
+        {
+            await sender.SendAsync(messages.Select(m => new MessageEnvelope(m, m.SessionId)), ct);
+        }
+
+        //await Parallel.ForEachAsync(
+        //    messageGroups,
+        //    ct,
+        //    async (messages, ct1) =>
+        //    {
+        //        this.Output.WriteLine("sending messages");
+        //        //using IServiceScope scope = this.TestServices.CreateScope();
+        //        //IMessageSender sender = scope.ServiceProvider.GetRequiredService<IMessageSender>();
+        //        await sender.SendAsync(messages.Select(m => new MessageEnvelope(m, m.SessionId)), ct1);
+        //    });
```

Comment:
Shouldn't this just be deleted? if not can you perhaps give a comment on when it becomes relevant again?

Reply:
I would like to reuse it, but it needs more work

Reply:
```suggestion
        // WIP - Yossi still developing approach
        //await Parallel.ForEachAsync(
        //    messageGroups,
        //    ct,
        //    async (messages, ct1) =>
        //    {
        //        this.Output.WriteLine("sending messages");
        //        //using IServiceScope scope = this.TestServices.CreateScope();
        //        //IMessageSender sender = scope.ServiceProvider.GetRequiredService<IMessageSender>();
        //        await sender.SendAsync(messages.Select(m => new MessageEnvelope(m, m.SessionId)), ct1);
        //    });
```